include/ignition/sensors/Sensor.hh

```diff
@@ -180,6 +189,14 @@ namespace ignition
       public: void AddSequence(ignition::msgs::Header *_msg,
                   const std::string &_seqKey = "default");
 
+      /// \brief Publishes information about the performance of the sensor.
+      ///        This method is called by Update().
+      ///        When different metrics need to be published this method
+      ///        could be overriden by subclasses.
+      /// \param[in] _now Current time.
+      public: virtual void PublishMetrics(
```

Comment:
Ahh the ABI checker is complaining about the addition of this virtual function:

https://build.osrfoundation.org/job/ignition_sensors-abichecker-any_to_any-ubuntu_auto-amd64/315/API_5fABI_20report/

Since no sensors are overriding this now, maybe we can keep it non virtual on Citadel~Edifice and make it virtual from Fortress.

Reply:
This will need to be virtual for the `CameraSensors` in order to be able to publish also `fps`  information.
So it will be needed shortly if it is expected to also fill the `fps` field of the message the sensor is a camera.

However, it is true that it isn't being overridden so to no break `abi` we could make it non-virtual or even private for the moment. Should I proceed that way?

Reply:
So... I actually had written a paragraph at the end of the previous comment that I deleted with some ideas to support `CameraSensor::PublishMetrics`... But then I remembered that we don't have APIs to get that from `ign-rendering` yet, see https://github.com/ignitionrobotics/ign-gazebo/issues/557#issuecomment-882851068. And adding that to `ign-rendering` will probably break ABI... So I think we'll need to target Fortress anyway.

Reply:
Ah makes sense. Well, I will remove `virtual` then to avoid abi break and then make it virtual from Fortress

Reply:
db8a8f5